From b9116d639c35fd7ad0bace000fe783e53f0ced17 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Thu, 15 Apr 2021 08:15:24 -0500
Subject: [PATCH 26/26] imx8mm: beacon: Update LPDDR4 RAM timings

In order to operate properly at extreme temperates, the LPDDR4 timings
need some updates.

Fixes: PHOENIX8-175
Signed-off-by: Adam Ford <aford173@gmail.com>
---
 board/beacon/imx8mm/lpddr4_timing.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/board/beacon/imx8mm/lpddr4_timing.c b/board/beacon/imx8mm/lpddr4_timing.c
index f3afa5dddf..e963b032db 100644
--- a/board/beacon/imx8mm/lpddr4_timing.c
+++ b/board/beacon/imx8mm/lpddr4_timing.c
@@ -16,16 +16,16 @@ struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d400304, 0x1 },
 	{ 0x3d400030, 0x1 },
 	{ 0x3d400000, 0xa1080020 },
-	{ 0x3d400020, 0x203 },
+	{ 0x3d400020, 0x202 },
 	{ 0x3d400024, 0x3a980 },
-	{ 0x3d400064, 0x5b00d2 },
+	{ 0x3d400064, 0x2d00d2 },
 	{ 0x3d4000d0, 0xc00305ba },
 	{ 0x3d4000d4, 0x940000 },
 	{ 0x3d4000dc, 0xd4002d },
 	{ 0x3d4000e0, 0x310000 },
 	{ 0x3d4000e8, 0x66004d },
 	{ 0x3d4000ec, 0x16004d },
-	{ 0x3d400100, 0x191e1920 },
+	{ 0x3d400100, 0x191e0c20 },
 	{ 0x3d400104, 0x60630 },
 	{ 0x3d40010c, 0xb0b000 },
 	{ 0x3d400110, 0xe04080e },
@@ -67,15 +67,15 @@ struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d400498, 0x620096 },
 	{ 0x3d40049c, 0x1100e07 },
 	{ 0x3d4004a0, 0xc8012c },
-	{ 0x3d402020, 0x1 },
+	{ 0x3d402020, 0x0 },
 	{ 0x3d402024, 0x7d00 },
 	{ 0x3d402050, 0x20d040 },
-	{ 0x3d402064, 0xc001c },
+	{ 0x3d402064, 0x6001c },
 	{ 0x3d4020dc, 0x840000 },
 	{ 0x3d4020e0, 0x310000 },
 	{ 0x3d4020e8, 0x66004d },
 	{ 0x3d4020ec, 0x16004d },
-	{ 0x3d402100, 0xa040305 },
+	{ 0x3d402100, 0xa040105 },
 	{ 0x3d402104, 0x30407 },
 	{ 0x3d402108, 0x203060b },
 	{ 0x3d40210c, 0x505000 },
@@ -92,7 +92,7 @@ struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d402194, 0x80303 },
 	{ 0x3d4021b4, 0x100 },
 	{ 0x3d4020f4, 0xc99 },
-	{ 0x3d403020, 0x1 },
+	{ 0x3d403020, 0x0 },
 	{ 0x3d403024, 0x1f40 },
 	{ 0x3d403050, 0x20d040 },
 	{ 0x3d403064, 0x30007 },
-- 
2.17.1

